x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ElementType.java

```diff
@@ -61,6 +63,10 @@ public static ElementType fromJava(Class<?> type) {
             elementType = DOUBLE;
         } else if (type == String.class || type == BytesRef.class) {
             elementType = BYTES_REF;
+        } else if (SpatialPoint.class.isAssignableFrom(type)) {
```

Comment:
This is very obscure to me but I read it like one type is mapped to one and only one type. This seems problematic with geo as depending how the data is consumed we might want to generate a different element type? I don't see how this is helping us on the generic solution and makes me very worried,

Reply:
I still think that when we re-enable doc values (next PR, I assume) this will be solvable. I've seen issues like this before, when I needed to pass down additional information to differentiate between types, and it was generally possible to do so, so I assume the same will be true here.

Reply:
I'm not a huge fan of `fromJava` in general. I think it's mostly useful when we parse constants and then want to convert them into types. And for tests. I suppose it's really useful in tests to be like "make me a block from these geometries", but, like you say, there are actually two choices of kinds of block. At least there are for points. Oh boy.

Reply:
Ir un the testa and this code seems never to be exercised. Please either add test to exercise the code or remove it. I think we should never have those classes here so I am more inclined to remove them.

Reply:
Removed support for SpatialType and Geometry. Turns out this was only excercised by tests that should themselves have been generating WKB, so I fixed the tests, and removed this code.